Add an option for photo refresh rate & a simple clock

ZachNagengast opened the pull request:
Hi! Found this repo while googling, though it was cool so added a few more features. Take a look and let me know what you think.

ZachNagengast committed:
Add refresh rate option

ZachNagengast committed:
Add simple clock 🕚

ZachNagengast committed:
Fix minor capitalization issue

jenil commented:
Hey @ZachNagengast 

Great work 👍 

Everything looks good, just that can we move the clock position to the bottom-center of the page and also reduce the size to about 50%? I want the center of the photo to get some visibility.

ZachNagengast commented:
Sure! I'm also updating the logic a bit so that it loads faster when a new image is ready to load in. I'll have a few more updates soon.

ZachNagengast commented:
There also might be an issue with unsplash api rate limits with the custom refresh interval option, I'm looking into that as well.

jenil commented:
I don't think the Unsplash API rate limit should be a problem, we have 5k rq/hr and we can request for more if needed.

ZachNagengast commented:
Cool, perhaps you can look at your usage and recommend a minimum refresh rate, I'm testing with refreshing the image every new tab and it goes quick.

jenil commented:
Here:
![image](https://user-images.githubusercontent.com/971411/26962397-848cf45c-4d03-11e7-8977-7c5001ee5d47.png)

It hardly goes over 6 request right now

ZachNagengast committed:
Preload images, update clock ui

ZachNagengast committed:
Merge branch 'master' of https://github.com/ZachNagengast/huntsplash

ZachNagengast commented:
Added image preloading, works nicely on my machine. Updated the clock too, looks better. I was getting some errors with the caching portion, maybe you can take a look, but it seems to still work. It loads and caches an extra image every time the current one updates, so it's always ahead by 1 and the user shouldn't see any loading. Normally this kind of thing should probably be done with a background.js script (I noticed http://instant.unsplash.com/ does this), but this method seems to work well enough. Open to any thoughts you had.

jenil commented on src/components/bg.vue:
```diff
@@ -57,52 +89,57 @@ export default {
-      if (this.BG.indexOf('data') === -1) {
-        this.$http.get(this.BG).then(response => {
+      if (cacheBG.indexOf('data') === -1) {
+        this.$http.get(cacheBG).then(response => {
```
Can you change this line to `this.$http.get(cacheBG, { responseType: 'blob' }).then(response => {`?
This will solve the error of caching the image.

ZachNagengast replied:
This worked, thanks!

ZachNagengast committed:
Fix cache typing issue, add greeting logic

ZachNagengast committed:
Remove option to refresh image every new tab

ZachNagengast commented:
Updated that issue with tracking, also added a greeting that changes based on the time of day.

jenil commented on src/components/clock.vue:
```diff
@@ -0,0 +1,87 @@
+</script>
+
+<style>
+.center {
```
Can you update the CSS here to?
```
    position: fixed;
    bottom: 0.5rem;
    left: 50%;
    -webkit-transform: translate(-50%, 0%);
```

jenil commented on src/components/clock.vue:
```diff
@@ -0,0 +1,87 @@
+.clock__hourtime {
+  display: inline-block;
+  color: #fff;
+  font-size: 5rem;
```
add `line-height: 1;`

jenil commented on src/components/clock.vue:
```diff
@@ -0,0 +1,87 @@
+.clock__greeting {
+  display: block;
+  color: #fff;
+  font-size: 3rem;
```
change font-size to 1.5rem

jenil reviewed (changes requested):
Hey @ZachNagengast, just review the changes and they work beautifully! I have left a few comments (minor CSS changes), once those are fixed I'll merge the PR :)

ZachNagengast committed:
Edit css per change request

ZachNagengast commented:
Cool, added!

jenil commented:
🥇 Awesome! Thank you :) 
Publishing 1.0.0 now... 🥂
